tests/api/controllers/robot_parallel_step/robot_parallel_step.c

```diff
@@ -74,29 +65,64 @@ int main(int argc, char **argv) {
 
   wb_robot_step_end();  // retrieve sensor values after step update
 
-  // step 3 - 96 ms -> check camera values after first pose update (second is only visible by the sensor after the next
+  // step 3 - 96 ms -> check sensor values after first pose update (second is only visible by the sensor after the next
   // step_end)
   wb_robot_step_begin(time_step);
 
-  image = wb_camera_get_image(camera);
+  // compass
+  values = wb_compass_get_values(compass);
+  const double newCompassValues1[3] = {0.0, 0.0, 1.0};
+  ts_assert_doubles_in_delta(3, values, newCompassValues1, 0.0001, "Compass measurement after first pose update is wrong.");
+
+  // distance sensor
+  distance = wb_distance_sensor_get_value(ds);
+  ts_assert_double_in_delta(distance, 650.0, 0.001, "Distance sensor measurement after first pose update is wrong.");
+
+  // gps
+  values = wb_gps_get_values(gps);
+  const double newGpsValues1[3] = {-0.05, 0.0295, 0.1};
+  ts_assert_doubles_in_delta(3, values, newGpsValues1, 0.00001, "GPS measurement after first pose update is wrong.");
 
-  // intense computation between begin and end
-  for (int i = 0; i < 1000000; i++) {
-    r = wb_camera_image_get_red(image, 1, 0, 0);
-    g = wb_camera_image_get_blue(image, 1, 0, 0);
-    b = wb_camera_image_get_green(image, 1, 0, 0);
+  // Give webots plenty of time to finish the step so we can be sure we'll detect any unexpected tampering with our values.
+  usleep(time_step * 1000 + 100000);
+
+  // Do some intensive computation, just for good measure
+  int result = 0xffff;
+  for (int i = 0; i < 0xffff; i++) {
+    result &= ~i;
   }
```

Comment:
[CS360](https://github.com/cyberbotics/webots/wiki/CPP-Coding-Style/#cs360-single-if-while-or-for-statement-must-be-written-without-brackets-the-statement-part-should-be-put-on-a-separate-line-in-case-of-nested-block-keep-brackets):
```suggestion
  for (int i = 0; i < 0xffff; i++)
    result &= ~i;
```